test/integration/targets/aws_api_gateway/tasks/main.yml

```diff
@@ -0,0 +1,182 @@
+- block:
+
+    # ============================================================
+    - name: test with no parameters
+      aws_api_gateway:
+      register: result
+      ignore_errors: true
+
+    - name: assert failure when called with no parameters
+      assert:
+        that:
+           - 'result.failed'
+           - 'result.msg.startswith("Region must be specified")'
+
+    # ============================================================
+    - name: test with minimal parameters but no region
+      aws_api_gateway:
+        api_id: 'fake-api-doesnt-exist'
+      register: result
+      ignore_errors: true
+
+    - name: assert failure when called with with minimal parameters but no region
+      assert:
+        that:
+           - 'result.failed'
+           - 'result.msg.startswith("Region must be specified")'
+
+    # ============================================================
+    - name: test disallow multiple swagger sources
+      aws_api_gateway:
+        api_id: 'fake-api-doesnt-exist'
+        region: 'fake_region'
+        swagger_file: foo.yml
+        swagger_text: "this is not really an API"
+      register: result
+      ignore_errors: true
+
+    - name: assert failure when called with with minimal parameters but no region
+      assert:
+        that:
+           - 'result.failed'
+           - 'result.msg.startswith("parameters are mutually exclusive")'
+
+    # This fails with
+
+    # msg": "There is an issue in the code of the module. You must
+    # specify either both, resource or client to the conn_type
+    # parameter in the boto3_conn function call"
+
+    # even though the call appears to include conn_type='client'
+
+    # # ============================================================
+    # - name: test invalid region parameter
+    #   aws_api_gateway:
+    #     api_id: 'fake-api-doesnt-exist'
+    #     region: 'asdf querty 1234'
+    #   register: result
+    #   ignore_errors: true
+
+    # - name: assert invalid region parameter
+    #   assert:
+    #     that:
+    #        - 'result.failed'
+    #        - 'result.msg.startswith("Region asdf querty 1234 does not seem to be available ")'
+
+    # ============================================================
+
+    - name: build API file 
+      template:
+        src: minimal-swagger-api.yml.j2
+        dest: "{{output_dir}}/minimal-swagger-api.yml"
+      tags: new_api,api,api_file
+
+    - name: deploy new API
+      aws_api_gateway: 
+        api_file: "{{output_dir}}/minimal-swagger-api.yml"
+        stage: "minimal"
+        region: '{{ec2_region}}'
```

Comment:
Out of curiosity, what is the default for this? Runs without specifying any ec2_region in test/integration/cloud-config-aws.yml.